src/generator/AutoRest.Go/Templates/MethodTemplate.cshtml

```diff
@@ -39,35 +39,69 @@
 @WrapComment("// ", sb.ToString())
 }
 
-func (client @(Model.Owner)) @(Model.MethodSignature) (@Model.MethodReturnSignature) {
-    @if (Model.ParameterValidations.Length > 0)
-    {
+func (client @(Model.Owner)) @(Model.MethodSignature) (@Model.MethodReturnSignature(false)) {
+@if (Model.IsLongRunningOperation())
+{
+    @:errChan:= make(chan error, 1)
+}
+@if (Model.ParameterValidations.Length > 0)
+{
     @:if err := validation.Validate([]validation.Validation{
-         @:@(Model.ParameterValidations)}); err != nil {
-             @:return result, @(Model.ValidationError)
-    @:}
-    @:@EmptyLine
+    @:@(Model.ParameterValidations)}); err != nil {
+    @if (Model.IsLongRunningOperation())
+    {
+    @:errChan <- validation.NewErrorWithValidationError(err, "redis.GroupClient", "Create")
+    @:close(errChan)
+    @:return nil, errChan
```

Comment:
Returning the nil channel will cause anybody reading it to block indefinitely. :) Demonstrated here: https://play.golang.org/p/09rkpDatlH

I'd refactor this so that the creation and deferral of closing them happens all before even this error handling.